Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,217 @@
+     */
+    Y.namespace('eZ');
+
+    var SELECTOR_CONTENT = '.dashboard-content';
+
+    /**
+     * The dashboard view
+     *
+     * @namespace eZ
+     * @class DashboardBlockBasedView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.DashboardBlockBasedView = Y.Base.create('dashboardBlockBasedView', Y.eZ.TemplateBasedView, [], {
+        initializer: function () {
+            this.after('activeChange', function () {
+                this.after('blocksChange', this._renderBlocks, this);
+
+                this._makeBlocksActive();
+            }, this);
+        },
+
+        /**
+         * Renders the dashboard view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockBasedView} the view itself
+         */
+        render: function () {
+            this.get('container').setHTML(this.template({dashboardTitle: this.get('dashboardTitle')}));
+            this._renderBlocks();
+
+            return this;
+        },
+
+        /**
+         * Activates all the blocks to allow communication between blocks and service
+         *
+         * @method _makeBlocksActive
+         * @protected
+         */
+        _makeBlocksActive: function () {
+            var blocks = this.get('blocks');
+
+            Object.keys(blocks).forEach(function (key) {
+                blocks[key].set('active', true);
+            });
+        },
+
+        /**
+         * Renders blocks on the dashboard
+         *
+         * @method _renderBlocks
+         * @protected
+         */
+        _renderBlocks: function () {
+            var fragment = Y.one(document.createDocumentFragment());
+
+            this._sortBlocks(this.get('blocks')).forEach(Y.bind(function (block) {
+                var targets = block.getTargets();
+
+                if (targets.indexOf(this) === -1) {
+                    block.addTarget(this);
+                }
+
+                fragment.append(block.render().get('container'));
+            }, this));
+
+            this.get('container').one(SELECTOR_CONTENT).setHTML(fragment);
+        },
+
+        /**
+         * Sorts the provided hash of blocks either by priority
+         * or alphabetically when both blocks have the same priority.
+         *
+         * @method _sortBlocks
+         * @protected
+         * @param blocks {Object} hash of blocks
+         * @return {Array} sorted list of blocks
+         */
+        _sortBlocks: function (blocks) {
+            var list = [];
+
+            Object.keys(blocks).forEach(function (key) {
+                list.push(blocks[key]);
+            });
+
+            list.sort(function (a, b) {
+                var priorityA = a.get('priority'),
+                    priorityB = b.get('priority');
+
+                if (priorityA === priorityB) {
+                    // sort alphabetically (from A to B)
+                    return a.get('identifier') < b.get('identifier');
+                } else {
```

Comment:
that's very picky and that also means more unit tests to cover that case. To be honest, I would not cover this case for now (we'll make sure to let enough room between the priority of our blocks)